SocialCareCaseViewerApi.Tests/V1/IntegrationTests/IntegrationTestHelpers.cs

```diff
@@ -0,0 +1,175 @@
+using System;
+using System.Collections.Generic;
+using Bogus;
+using Microsoft.EntityFrameworkCore;
+using SocialCareCaseViewerApi.V1.Boundary.Requests;
+using SocialCareCaseViewerApi.V1.Infrastructure;
+using DbPerson = SocialCareCaseViewerApi.V1.Infrastructure.Person;
+
+namespace SocialCareCaseViewerApi.Tests.V1.IntegrationTests
+{
+    public static class IntegrationTestHelpers
+    {
+        public static DbPerson CreateExistingPerson(DatabaseContext context, int? personId = null, string ageContext = null, string restricted = null)
+        {
+            var person = new Faker<DbPerson>()
+                .RuleFor(p => p.Id, f => personId ?? f.UniqueIndex + 1)
+                .RuleFor(p => p.Title, f => f.Name.Prefix())
+                .RuleFor(p => p.FirstName, f => f.Person.FirstName)
+                .RuleFor(p => p.LastName, f => f.Person.FirstName)
+                .RuleFor(p => p.FullName, f => f.Person.FullName)
+                .RuleFor(p => p.DateOfBirth, f => f.Person.DateOfBirth)
+                .RuleFor(p => p.DateOfDeath, f => f.Date.Recent())
+                .RuleFor(p => p.Ethnicity, f => f.Commerce.Color())
+                .RuleFor(p => p.FirstLanguage, f => f.Random.Word())
+                .RuleFor(p => p.Religion, f => f.Random.String2(1, 5))
+                .RuleFor(p => p.EmailAddress, f => f.Person.Email)
+                .RuleFor(p => p.Gender, f => f.Random.String2(1, "MF"))
+                .RuleFor(p => p.Nationality, f => f.Address.Country())
+                .RuleFor(p => p.NhsNumber, f => f.Random.Number(int.MaxValue))
+                .RuleFor(p => p.PersonIdLegacy, f => f.Random.String2(16))
+                .RuleFor(p => p.AgeContext, f => ageContext ?? f.Random.String2(1, "AC"))
+                .RuleFor(p => p.DataIsFromDmPersonsBackup, f => f.Random.String2(1, "YN"))
+                .RuleFor(p => p.SexualOrientation, f => f.Random.String2(10))
+                .RuleFor(p => p.PreferredMethodOfContact, f => f.Random.String2(10))
+                .RuleFor(p => p.Restricted, f => restricted ?? f.Random.String2(1, "YN")).Generate();
+
+            var insertPersonQuery = SeedPerson(person);
+
+            context.Database.ExecuteSqlRaw(insertPersonQuery);
```

Comment:
Is there a reason we have to use these raw SQL strings to insert a person into the DB? Could we not use the DatabaseGateWay and take advantage of EF? I might be missing something but this seems like a more complex way of doing this. If our Person changes then we need to update our SQL for the tests.

Reply:
The reason for using the `ExecuteSqlRaw()` was so that we don't have the query triggering a transaction.

One of the issues was getting the test database populated with the data. After asking around a bit, it was suggested that the reason the data wasn't getting to our database was because we were using transactions and for those changes to save to the database we needed to commit the transaction and it turns out when we call `Database.SaveChanges()` in our actual code, it doesn't commit the transaction the integration test. 🤷🏾‍♀️ 

Adding `transaction.Commit()` saved the data to the DB but it didn't replicate the bug when running the test. 

It's still unclear to me but happy to dig into this a bit better later but it was returning a false positive as well and would replace the `workerteam` data in the test database rather than add to it which is what we're trying to replicate it. 

Based on all the stuff we'd researched to understand what might be going on, my hypothesis was that if we seeded the test database in way that entity framework doesn't know about the data initially, we could then run the rest of the code using entity framework, and this replicated the bug.

The hypothesis stems from the fact that we'd figured that there was something different in the setup of our test database compared to the local dev database since our tests followed the user journey properly. My current understanding is the possibility of some magic happening when Entity Framework is tracking entities that is not entirely reflected in the actual database.

But happy to try and dig into this a bit further.

Reply:
We've had issues with tracking in gateway tests before and have disabled it in certain tests. That might help in your use case as well. So when you initialise your DatabaseContext you can disable the tracker:  `DatabaseContext.ChangeTracker.QueryTrackingBehavior = QueryTrackingBehavior.NoTracking;`

Reply:
@LBHTKarki Yeah, Ting mentioned this and when we tried it the data didn't update. Checking the [docs](https://docs.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.dbcontextoptionsbuilder.usequerytrackingbehavior?view=efcore-5.0) we found that we shouldn't disable tracking if we want to manipulate and persist changes.

Reply:
@Renny4Real yes, disabling it for the actual implementation is not a good idea and we use it for automated audit for example. It just sometimes useful in tests when you are dealing with related entities. For example when you create related entities your tests might be passing even if your gateway queries are not setup correctly. You'll only notice the issue at runtime. Tracking combined with tansactions in tests can result in some challening situations like this one. Glad you are tackling integration tests and this setup is really useful for future tests